infrastructure/storage/src/lmdb_store/store.rs

```diff
@@ -133,10 +133,37 @@ impl LMDBBuilder {
             builder.open(&path, flags, 0o600)?
         };
         let env = Arc::new(env);
+
+        // Increase map size if usage gets close to the db size
+        let mut env_info = env.info()?;
+        let env_stat = env.stat()?;
+        let size_used = env_stat.psize as usize * env_info.last_pgno;
+        let mut space_remaining = env_info.mapsize - size_used;
+        let usage = (size_used as f64 / env_info.mapsize as f64) * 100.0;
+        if space_remaining <= ((self.db_size_mb * 1024 * 1024) as f64 * 0.5) as usize {
+            unsafe {
+                env.set_mapsize(size_used + self.db_size_mb * 1024 * 1024)?;
```

Comment:
Just noting that all the caveats/suggestions for using [`set_mapsize`](https://docs.rs/lmdb-zero/0.4.4/lmdb_zero/struct.Environment.html#method.set_mapsize) seem to be adhered to, namely:

- It's set to a multiple of the original database size which is a multiple of 1MiB (presumably what "OS page size" means)
- There won't be any active db transactions at this point in the code

:+1: